### PR TITLE
Set improvements

### PR DESCRIPTION
**Pull request by will**
Documentation, giving `#to_set` to all `Enumerable`, some refactoring, a few new specs, and returning `self` in methods that ruby does.

### CODE REVIEW

File: src/set.cr [hunk 1]
```diff
@@ -1,52 +1,122 @@
...
+  #     s = Set.new [1,5]
+  #     s.includes? 8     # => false
+  #     s << 8
+  #     s.includes? 8     # => true
   def add(object : T)
     @hash[object] = nil
+    self
   end
 
+  # Adds `#each` element of `elms` to the set and returns `self`
```
Comment (jhass) [inline src/set.cr]: "elems"

File: src/set.cr [hunk 2]
```diff
@@ -110,32 +195,48 @@ struct Set(T)
...
     io << "}"
   end
 
+  # Returns `true` if the set is a subset of the `other` set
+  #
+  # This set must have fewer elements than the `other` set, and all of elements
+  # in this set must be present in the `other` set.
+  #
+  #     Set.new([1,5]).subset? Set.new([1,3,5]) # => true
   def subset?(other : Set)
```
Comment (jhass) [inline src/set.cr]: Maybe we should rename this to `strict_subset?` and add a `subset?` that allows equality? https://en.wikipedia.org/wiki/Subset

Comment (will) [inline src/set.cr]: It looks like ruby has it as [`proper_subset`](http://ruby-doc.org/stdlib-2.2.3/libdoc/set/rdoc/Set.html#method-i-proper_subset-3F). I've not actually used these methods, but it probably does make sense to add them

**Comment by will** [comment]
Added the proper_versions and fixed the elms thing

**Comment by jhass** [comment]
Thank you!

**Pull request merged**